syntax = "proto3";

package todoPpackage;

service Todo {
    rpc createTodo(TodoItem) returns (TodoItem);
    rpc readTodos(no_param) returns (TodoItems);
}

// Can't leave empty input arg, must specify a message for void/empty/or whatevs
message no_param {}

message TodoItem {
    int32 id = 1;
    string text = 2;
}

message TodoItems{
    repeated TodoItem items = 1;
}